src/main/java/core/basesyntax/ShapeSupplier.java

```diff
@@ -0,0 +1,39 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class ShapeSupplier {
+    private final int defaultSize = 20;
+    private final int maxSize = 100;
+    private final int maxBound = 10;
+    private ColorSupplier colorSupplier = new ColorSupplier();
+    private Random randomizer = new Random();
+
+    private int randomSize() {
+        return 1 + randomizer.nextInt();
```

Comment:
The `randomSize` method currently uses `randomizer.nextInt()` without specifying a bound, which can lead to unexpected results. It should be `randomizer.nextInt(maxSize)` to ensure the size is within a reasonable range.